packages/cozy-client/src/flagship-certification/flagship-certification.spec.js

```diff
@@ -0,0 +1,235 @@
+import { certifyFlagship } from './flagship-certification'
+import { createMockClient } from '../mock'
+
+import { getAppAttestationFromStore } from './store-attestation'
+
+jest.mock('./store-attestation', () => ({
+  getAppAttestationFromStore: jest.fn()
+}))
+
+const getClientMock = () => {
+  const client = createMockClient({
+    clientOptions: {
+      oauth: {}
+    }
+  })
+
+  client.stackClient.oauthOptions = {
+    clientID: 'SOME_CLIENT_ID',
+    registrationAccessToken: 'SOME_REGISTRATION_ACCESS_TOKEN'
+  }
+
+  return client
+}
+
+const mockCorrectChallengeRequest = client => {
+  client.stackClient.fetchJSON.mockResolvedValueOnce({
+    nonce: 'SOME_NONCE'
+  })
+}
+
+const mockCorrectStoreApiRequest = () => {
+  getAppAttestationFromStore.mockResolvedValueOnce({
+    platform: 'android',
+    attestation: 'SOME_STORE_ATTESTATION',
+    keyId: 'SOME_KEY_ID'
+  })
+}
+
+const mockCorrectCertificationConfig = () => {
+  return {
+    androidSafetyNetApiKey: 'SOME_ANDROID_SAFETY_NET_API_KEY'
+  }
+}
+
+describe('certifyFlagship', () => {
+  beforeEach(() => {
+    jest.spyOn(console, 'warn').mockImplementation(() => jest.fn())
+  })
+
+  afterEach(() => {
+    jest.restoreAllMocks()
+    getAppAttestationFromStore.mockReset()
+  })
+
+  it('should correctly call certification endpoints', async () => {
+    const client = getClientMock()
+
+    const certificationConfig = mockCorrectCertificationConfig()
+
+    mockCorrectChallengeRequest(client)
+
+    mockCorrectStoreApiRequest()
+
+    await certifyFlagship(certificationConfig, client)
+
+    // Should ask the stack for a challenge
+    expect(client.stackClient.fetchJSON).toHaveBeenCalledWith(
+      'POST',
```

Comment:
let's  refactor

expect(client.stackClient.fetchJSON).toHaveBeenCalledWith(fetchJSONExpectedParams(url))

in all the following tests 🙏

Reply:
What about this? https://github.com/cozy/cozy-client/compare/f99251f97be750c0363bcd659d1628c5937602de..d9ea74471e1e23ed356a9e710722a2939f4b6972